src/common/mfu_flist_remove.c

```diff
@@ -588,6 +589,54 @@ void mfu_flist_unlink(mfu_flist flist)
     int levels, minlevel;
     mfu_flist* lists;
     mfu_flist_array_by_depth(flist, &levels, &minlevel, &lists);
+    mfu_flist pstatlist;
+
+    /* if traceless, dump the stat of each item's pdir */
+    if (traceless) {
+        uint64_t idx;
+        /* get number of elements in our local list */
+        uint64_t size = mfu_flist_size(flist);
+
+        pstatlist = mfu_flist_new();
+
+        flist_t* file_list = (flist_t*)pstatlist;
+
+        /* we will stat all items in output list, so set detail to 1 */
+        file_list->detail = 1;
+
+        /* get user data if needed */
+        if (file_list->have_users == 0) {
+            mfu_flist_usrgrp_get_users(pstatlist);
+        }
+
+        /* get groups data if needed */
+        if (file_list->have_groups == 0) {
+            mfu_flist_usrgrp_get_groups(pstatlist);
+        }
+
```

Comment:
We've got a new mfu_flist_set_detail(flist, 1) function that we can use here to replace lines 602-615.